.github/workflows/ci.yml

```diff
@@ -9,7 +9,7 @@ jobs:
   build-and-test:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest, macos-14, windows-latest]
```

Comment:
I like keeping both (and maybe adding ubuntu-arm). Go doesn't really need it because we can trust the runtime across CPU arch's, but I wonder if it's a good practice to have all CIs of Temporal-distributed software test against ubuntu x64/arm, max x64/arm, and windows x64. Your call.